pkg/uio/reader_test.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package uio
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+)
+
+func readAndCheck(t *testing.T, want, tmpfileP string) {
+	t.Helper()
+	r := strings.NewReader(want)
+	if err := ReadIntoFile(r, tmpfileP); err != nil {
+		t.Errorf("ReadIntoFile(%v, %s) = %v, want no error", r, tmpfileP, err)
+	}
+
+	got, err := ioutil.ReadFile(tmpfileP)
+	if err != nil {
+		t.Fatalf("ioutil.ReadFile(%s) = %v, want no error", tmpfileP, err)
+	}
+	if want != string(got) {
+		t.Errorf("got: %v, want %s", string(got), want)
+	}
+}
+
+func TestReadIntoFile(t *testing.T) {
+	want := "I am the wanted"
+
+	// Write to a file already exist.
+	tmpfile, err := ioutil.TempFile("", "uio-out")
```

Comment:
I am sorry, I gave you the wrong name.

For this test, I would suggest
dir, err :=  MkdirTemp("", "uio-out")
if err etc. 
defer os.RemoveAll(dir)

ok, now you have a directory you own, reasonably unique.

From this point forward:
os.CreateFile(filepath.Join(dir, "wahtever"))
and if the file should not exist
file.path.Join(dir, "notexisting")

can be the name. But it is unsafe to remove a file by name and assume it will not come back. 

Sorry I was not informative enough.

Reply:
Q: is the point being that removing directory being more reliable than removing single file ? 

the file can come back argument applies to directory also ?

Reply:
Done.

Reply:
Done.